Marketplace/voiceHUDApp/appResources/appData/voiceHUD_app.js

```diff
@@ -0,0 +1,836 @@
+        if (user.audioLevel > 0.25){ 
+            user.hudID.forEach(function(id){
+                Overlays.editOverlay(id, { visible: true });
+            });
+            user.audioBarIDs.forEach(function(id){
+                Overlays.editOverlay(id, { visible: true });
+            });
+        } else {
+            setDefaultHUD(uuid);
+            Script.setTimeout(function(){
+                user.hudID.forEach(function(id){
+                    if (user.avgAudioLevel < 0.25){
+                        Overlays.editOverlay(id, { visible: false });
+                    }
+                });
+                user.audioBarIDs.forEach(function(id){
+                    if (user.avgAudioLevel < 0.25){
+                        Overlays.editOverlay(id, { visible: false });
+                    }
+                });
+            }, 5000);
+        }
+        if (user.audioLevel <= 0.5){ 
+            user.audioBarIDs.forEach(function(id){
+                Overlays.editOverlay(id, { color: {
+                    "red": 0,
+                    "green": 255,
+                    "blue": 0
+                } });
+            });
+        } else if (user.audioLevel > 0.5 && user.audioLevel <= 0.75){ 
+            user.audioBarIDs.forEach(function(id){
+                Overlays.editOverlay(id, { color: {
+                    "red": 255,
+                    "green": 255,
+                    "blue": 0
+                } });
+            });
+        } else if (user.audioLevel > 0.75){ 
+            user.audioBarIDs.forEach(function(id){
+                Overlays.editOverlay(id, { color: {
+                    "red": 255,
+                    "green": 0,
+                    "blue": 0
+                } });
+            });
+        }
+        user.audioBarIDs.forEach(function(index){                
+            var dimensionsWithSound = HUD_MIN_Z_DIMENSIONS + user.audioLevel * 
+                (HUD_MAX_Z_DIMENSIONS[user.audioBarIDs.indexOf(index)] - HUD_MIN_Z_DIMENSIONS);
+            Overlays.editOverlay(index, { 
+                dimensions: { x: HUD_DEFAULT_DIMENSIONS.x, y: HUD_DEFAULT_DIMENSIONS.y, z: dimensionsWithSound } 
+            });
+        });            
+    }
+
+
+    // Set the bars of the HUD to the default size
+    function setDefaultHUD(uuid) {
+        var user = userStore[uuid];
+        user.audioBarIDs.forEach(function(id){
+            Overlays.editOverlay(id, { dimensions: HUD_DEFAULT_DIMENSIONS });
+        });   
+    }
+
+
+    // The function removes all HUDs from all users
+    function removeAll() {
+        // remove previous overlays
+        for (var i = 0; i < settings.users.length; i++) {
+            var user = settings.users[i];
+            var uuid = user.uuid;
+            deleteHUD(uuid);
+        }
+        var overlayList = Overlays.findOverlays(MyAvatar.position, 1000);
+        overlayList.forEach(function(overlay){
+            if (Overlays.getProperty(overlay, "name") === "Voice-HUD"){
```

Comment:
ah, ok, I understand why you named all of the overlays `Voice-HUD` now.

this method isn't super scalable. Is there a reason you aren't keeping track of each overlay ID associated with each user somewhere? i.e. `settings.users` ?

Reply:
Overlay IDs are stored per user in user.hudID and user.audioBarIDs but for whatever reason, they can get stuck or otherwise missed for deletion.